src/main/java/com/matt_richardson/gocd/websocket_notifier/PipelineDetailsPopulator.java

```diff
@@ -12,15 +12,20 @@
 import java.net.HttpURLConnection;
 import java.net.URL;
 
+import org.apache.commons.codec.binary.Base64;
+
 public class PipelineDetailsPopulator {
     private static Logger LOGGER = Logger.getLoggerFor(GoNotificationPlugin.class);
+    private static PluginConfig pluginConfig;
     private int httpPort;
 
     public PipelineDetailsPopulator() {
-        this.httpPort = 8153;
+        pluginConfig = new PluginConfig();
+        this.httpPort = pluginConfig.getGoHttpPort();
     }
 
     public PipelineDetailsPopulator(int httpPort) {
+        pluginConfig = new PluginConfig();
```

Comment:
Yep. It can sometimes get a bit messy, but its preferable to pass dependencies like the config around rather than new up objects.